### PR TITLE
Events

### PR DESCRIPTION
**Pull request by AlejandroNeira98**
# The main changes in this part of the project were:
- Copy  the files `calculate.js` and `operate.js` into a `logic/` directory in your project.
- Analyze the files `calculate.js` and `operate.js`, be sure you understand what they do (note: you need to add [big.js](https://www.npmjs.com/package/big-js) as dependency).
- Import the files in your Calculator component.
- Implement the event handlers you need to use the math logic from `calculate.js` and `operate.js` in your React component.
- Practice what you have learned until now: use `state` and `props`, pass props to *child* components, use lifecycle methods, *lift state up*.
- Test your app in the browser. It should be able to perform math operations.

### CODE REVIEW

File: src/components/Calculator.js [hunk 1]
```diff
@@ -1,39 +1,84 @@
...
+    } else if (symbol === '=') {
+      this.setState(calculate({ total, next, operation }, symbol));
+      this.setState({ display: total });
+    } else {
+      this.setState(calculate({ total, next, operation }, symbol));
+      this.setState({ display: total });
+    }
+    console.log(this.state);
+  }
+
```
Comment (mmsesay) [inline src/components/Calculator.js]: - [ ] The calculations are not working as expected. Please take a look at the gif below to understand the point am trying to make here.

Gif of Demo Test|
:--------------------:|
math

I will suggest you work on ensuring that the calculator's result is accurate, and can perform all math operations. 

Also, try making use of lifecycle methods.  I think they might be useful in these operations.

File: src/components/Calculator.js [hunk 2]
```diff
@@ -1,39 +1,84 @@
...
+          <Key symbol="3" retrieveSymbol={this.retrieveSymbol} />
+          <Key symbol="+" color="orange" retrieveSymbol={this.retrieveSymbol} />
         </div>
         <div className="row">
-          <Key symbol="0" color="double" />
-          <Key symbol="." />
-          <Key symbol="=" color="orange" />
+          <Key symbol="0" color="double" retrieveSymbol={this.retrieveSymbol} />
+          <Key symbol="." retrieveSymbol={this.retrieveSymbol} />
+          <Key symbol="=" color="orange" retrieveSymbol={this.retrieveSymbol} />
```
Comment (mmsesay) [inline src/components/Calculator.js]: - [OPTIONAL] I think you might need to a re-arrange your  UI. The calculator design is all the way to the top left of the screen.

See screenshot|
:---------------------------:|
a1

**Review by mmsesay — CHANGES_REQUESTED** [review]
Hi @AlejandroNeira98, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Meltrust — APPROVED** [review]
Hi @AlejandroNeira98,


## Wow, you did it 🎉 It works!
Brilliant
## Thank you for correcting the calculator functionality, as requested by the previous reviewer 💪 🥇 ㊗️ 
### Well done!




Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**